src/goto-symex/symex_goto.cpp

```diff
@@ -65,6 +67,167 @@ void goto_symext::apply_goto_condition(
   }
 }
 
+/// Try to evaluate a simple pointer comparison.
+/// \param operation: ID_equal or ID_not_equal
+/// \param symbol_expr: The symbol expression in the condition
+/// \param other_operand: The other expression in the condition - must pass
+///   goto_symex_is_constant, and since it is pointer-typed it must therefore
+///   be an address of expression, a typecast of an address of expression or a
+///   null pointer
+/// \param value_set: The value-set for looking up what the symbol can point to
+/// \param language_mode: The language mode
+/// \param ns: A namespace
+/// \return If we were able to evaluate the condition as true or false then we
+///   return that, otherwise we return an empty optionalt
+static optionalt<renamedt<exprt, L2>> try_evaluate_pointer_comparison(
+  irep_idt operation,
+  const symbol_exprt &symbol_expr,
+  const exprt &other_operand,
+  const value_sett &value_set,
+  const irep_idt language_mode,
+  const namespacet &ns)
+{
+  const exprt *other_operand_with_typecasts_removed = &other_operand;
+  const typecast_exprt *typecast_expr = expr_try_dynamic_cast<typecast_exprt>(
+    *other_operand_with_typecasts_removed);
+
+  while(typecast_expr)
+  {
+    other_operand_with_typecasts_removed = &typecast_expr->op();
+    typecast_expr = expr_try_dynamic_cast<typecast_exprt>(
+      *other_operand_with_typecasts_removed);
+  }
```

Comment:
Use `skip_typecast`?

Reply:
I tried to search for such a utility but didn't hit the right combination of words. Next time I'll try looking in `expr_utils.h`.